os/arch/arm/src/armv7-m/up_memfault.c

```diff
@@ -121,6 +122,8 @@ int up_memfault(int irq, FAR void *context, FAR void *arg)
 	uint32_t *regs = (uint32_t *)context;
 	uint32_t cfsr = getreg32(NVIC_CFAULTS);
 	uint32_t mmfar = getreg32(NVIC_MEMMANAGE_ADDR);
+	g_assertpc = regs[REG_R15];
```

Comment:
In case of IACCVIOL, the PC value might be invalid. So let us set R14 for that case.

Reply:
Done